aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,223 @@
+
+Configuring the JSON Formatter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A common request when formatting responses using JSON is to use JavaScript-style "camel" casing (the first character of each property is lowercase, then uppercase is used to begin each additional word making up the name). Support for camel casing can be added by using the ``AddJsonOptions`` extension method when configuring MVC:
+
+.. code-block:: c#
+
+  services.AddMvc()
+    .AddJsonOptions(o =>
+      o.SerializerSettings.ContractResolver = 
+        new CamelCasePropertyNamesContractResolver());
+
+Replace the default ``SerializerSettings.ContractResolver`` with a new instance of ``CamelCasePropertyNamesContractResolver``. Before this change, an object with a property of "FirstName" and a value of "Steve" would be serialized to JSON as ``{"FirstName":"Steve"}``; after the change to use camel case, it would serialize as ``{"firstName":"Steve"}``.
+
+Adding XML Format Support
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To add support for XML formatting, add the "Microsoft.AspNetCore.Mvc.Formatters.Xml" package to your *project.json*'s list of dependencies.
+
+Add the XmlSerializerFormatters to MVC's configuration in *Startup.cs*:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Startup.cs
+  :language: c#
+  :lines: 30-36
+  :emphasize-lines: 4
+  :dedent: 8
+
+Alternately, you can add just the output formatter:
+
+.. code-block:: c#
+
+  services.AddMvc(options =>
+  {
+    options.OutputFormatters.Add(new XmlSerializerOutputFormatter());
+  });
+
+Once you've added support for XML formatting, your API methods should return the appropriate format based on the request's ``Accept`` header, as this Fiddler example demonstrates:
+
+.. image:: formatting/_static/xml-response.png
+
+You can see in the Inspectors tab that the Raw GET request was made with an ``Accept: application/xml`` header set. The response pane shows the ``Content-Type: application/xml`` header, and the ``Author`` object has been serialized to XML.
+
+Use the Composer tab to modify the request to specify ``application/json`` in the ``Accept`` header. Execute the request, and the response will be formatted as JSON:
+
+.. image:: formatting/_static/json-response-fiddler.png
+
+In this screenshot, you can see the request sets a header of ``Accept: application/json`` and the response specifies the same as its ``Content-Type``. The ``Author`` object is shown in the body of the response, in JSON format.
+
+Forcing a Particular Format
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you would like to require the use of a particular response format, without resorting to using a return type like ``JsonResult``, you can apply the ``[Produces]`` filter. The ``[Produces]`` filter specifies a response format. Like most :doc:`/mvc/controllers/filters`, this can be applied at the action, controller, or global scope.
```

Comment:
> If you would like to require the use of a particular response format, without resorting to using a return type like ``JsonResult``, you can

If you would like to restrict the response formats for a specific action you can